// 版本
syntax = "proto3";
// 类似于C++中的namespace
package RPC;
// 生成 service类和描述rpc方法
option cc_generic_services = true;

// 响应码
message ResultCode
{
    bytes errmsg = 1;
    int32 errcode = 2;
}

// 登录请求消息
message LoginRequest
{
    bytes name = 1;
    bytes pwd = 2;
}

// 登录响应消息
message LoginResponse
{
    ResultCode code = 1;
    bool success = 2;
}

//注册的请求消息
message RegisterRequest
{
    int32 id = 1;
    bytes name = 2;
    bytes pwd = 3;
}

// 注册的响应消息
message RegisterResponse
{
    ResultCode rescode = 1;
    bool success = 2;
}

// 生成 service类 和rpc调用方法
service UserServiceRpc
{
    rpc Login(LoginRequest) returns (LoginResponse);

    rpc Register(RegisterRequest) returns(RegisterResponse);

}